src/client/Client.cc

```diff
@@ -5825,7 +5826,8 @@ void Client::unmount()
 {
   Mutex::Locker lock(client_lock);
 
-  assert(mounted);  // caller is confused?
+  if (!mounted)
```

Comment:
Indeed. I changed a couple of places to return errors here instead of asserting, as that's something we really want to avoid in libraries, unless we have no other choice.